runtime/near-vm-errors/src/lib.rs

```diff
@@ -130,9 +154,10 @@ pub enum CompilationError {
 /// Error that can occur while preparing or executing Wasm smart-contract.
 pub enum PrepareError {
     /// Error happened while serializing the module.
-    Serialization,
+    Serialization(StringError),
     /// Error happened while deserializing the module.
-    Deserialization,
+    Deserialization(StringError),
```

Comment:
>This is indended to be NFC in that there should not be any meaningful functionality
differences other than a change the error messages and variants seen.

Alas, that's not the case, and this is one example. The following code works on master, but fails with this PR:

```rust
#[test]
fn serialization_compat() {
    let _: PrepareError = BorshDeserialize::try_from_slice(&[1]).unwrap();
}
```

And this `[1]` is stored in the database for real transactions.